src/ess/powder/nexus.py

```diff
@@ -0,0 +1,251 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+
+"""NeXus input/output for DREAM.
+
+Notes on the detector dimensions (2024-05-22):
+
+See https://confluence.esss.lu.se/pages/viewpage.action?pageId=462000005
+and the ICD DREAM interface specification for details.
+
+- The high-resolution and SANS detectors have a very odd numbering scheme.
+  The scheme attempts to follows some sort of physical ordering in space (x,y,z),
+  but it is not possible to reshape the data into all the logical dimensions.
+"""
+
+import warnings
+from typing import Any
+
+import scipp as sc
+import scippnexus as snx
+from ess.reduce import nexus
+
+from ess.powder.types import (
+    DetectorBankSizes,
+    DetectorEventData,
+    Filename,
+    MonitorEventData,
+    MonitorType,
+    NeXusDetector,
+    NeXusDetectorName,
+    NeXusMonitor,
+    NeXusMonitorName,
+    RawDetector,
+    RawMonitor,
+    RawMonitorData,
+    RawSample,
+    RawSource,
+    ReducibleDetectorData,
+    RunType,
+    SamplePosition,
+    SourcePosition,
+)
+
+
+def load_nexus_sample(file_path: Filename[RunType]) -> RawSample[RunType]:
+    return RawSample[RunType](nexus.load_sample(file_path))
+
+
+def dummy_load_sample(file_path: Filename[RunType]) -> RawSample[RunType]:
+    """
+    In test files there is not always a sample, so we need a dummy.
+    """
+    return RawSample[RunType](
+        sc.DataGroup({'position': sc.vector(value=[0, 0, 0], unit='m')})
+    )
+
+
+def load_nexus_source(file_path: Filename[RunType]) -> RawSource[RunType]:
+    return RawSource[RunType](nexus.load_source(file_path))
+
+
+def load_nexus_detector(
+    file_path: Filename[RunType], detector_name: NeXusDetectorName
+) -> NeXusDetector[RunType]:
+    definitions = snx.base_definitions()
+    definitions["NXdetector"] = FilteredDetector
+    # Events will be loaded later. Should we set something else as data instead, or
+    # use different NeXus definitions to completely bypass the (empty) event load?
```

Comment:
You used this already for monitors. Why not here?

Reply:
Because ScippNexus raises when trying to use selection with a dimension or field (`event_time_zero`) which does not exist. The NXevent_data groups in the monitors of our DREAM file are completely empty. I did not see problems with the detectors, but I can double check why it works there...